packages/firebase_ui_firestore/lib/src/query_builder.dart

```diff
@@ -502,3 +502,48 @@ class FirestoreListView<Document> extends FirestoreQueryBuilder<Document> {
           },
         );
 }
+
+/// Listens to an aggregate query and passes the [AsyncSnapshot] to the builder.
+class AggregateQueryBuilder extends StatefulWidget {
+  /// A query to listen to
+  final AggregateQuery query;
+
+  /// A builder that is called whenever the query is updated.
+  final Widget Function(
+    BuildContext context,
+    AsyncSnapshot<AggregateQuerySnapshot> snapshot,
+  ) builder;
+
+  const AggregateQueryBuilder({
+    super.key,
+    required this.query,
+    required this.builder,
+  });
+
+  @override
+  State<AggregateQueryBuilder> createState() => _AggregateQueryBuilderState();
+}
+
+class _AggregateQueryBuilderState extends State<AggregateQueryBuilder> {
+  late var queryFuture = widget.query.get();
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<AggregateQuerySnapshot>(
+      key: ValueKey(queryFuture),
```

Comment:
Why is this here?

Reply:
to get rid of redundant FutureBuilder rebuilds on hot reload (reassemble)

Reply:
This has the nasty effect of destroying the entire subtree whenever the query changes

I don't quite understand which redundant rebuild you're referring to.
But this possibly negatively impacts production code